OpenLocoTool/Annotation.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenLocoTool
+{
+	public class Annotation
+	{
+		private string name = "";
+		private Annotation? parent = null;
+		private int start = 0;
+		private int end = 0;
+		private int length = 0;
+
+		public Annotation(string name, Annotation? parent, int start, int length)
+		{
+			Name = name;
+			Parent = parent;
+			this.start = start;
+			Length = length;
+		}
+
+		public Annotation(string name, int start, int length)
+		{
+			Name = name;
+			this.start = start;
+			Length = length;
+		}
+
+		public string Name { get => name; set => name = value; }
+		public Annotation? Parent { get => parent; set => parent = value; }
+		public int Start { get => start; set { start = value; Length = Length; } }
```

Comment:
bug here, should be `Length = length`, and fwiw you can use autoproperties for all this or even records, it'll save having to write so much code here

Reply:
actually maybe not

Reply:
Changed to Length = length. Part of the idea is that setting Start to a new value keeps the length the same and recalculates the End value. Not necessarily that important.